PROTO_tests/tests/dtypes_test.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+from arkouda import dtypes
+
+"""
+DtypesTest encapsulates arkouda dtypes module methods
+"""
+
+SUPPORTED_NP_DTYPES = [bool, int, float, str, np.int64, np.float64, np.uint8, np.uint64, np.str_]
+
+
+class TestDTypes:
+    @pytest.mark.parametrize("dtype", SUPPORTED_NP_DTYPES)
+    def test_check_np_dtype(self, dtype):
+        """
+        Tests dtypes.check_np_dtype method
+
+        :return: None
+        :raise: AssertionError if 1.. test cases fail
+        """
+        dtypes.check_np_dtype(np.dtype(dtype))
+
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype(np.dtype(np.int16))
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("np.str")
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("bigint")
+
+    @pytest.mark.parametrize(
+        "dtype, byte_size, kind",
+        [
+            (np.bool_, 1, "bool"),
+            (bool, 1, "bool"),
+            (np.int64, 8, "int"),
+            (np.float64, 8, "float"),
+            (np.uint8, 1, "uint"),
+            (np.str_, 0, "str"),
+            (str, 0, "str"),
+        ],
+    )
+    def test_translate_np_dtype(self, dtype, byte_size, kind):
+        """
+        Tests dtypes.translate_np_dtype method
+
+        :return: None
+        :raise: AssertionError if 1.. test cases fail
+        """
+        d_tuple = dtypes.translate_np_dtype(np.dtype(dtype))
+        assert kind == d_tuple[0]
+        assert byte_size == d_tuple[1]
+
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype(np.dtype(np.int16))
+        with pytest.raises(TypeError):
+            dtypes.translate_np_dtype("np.str")
+
+    @pytest.mark.parametrize(
+        "dtype, val",
+        [
+            ("bool", True),
+            ("int64", 1),
+            ("float64", float(0.0)),
+            ("str", "test"),
+            ("int64", np.int64(1)),
+            ("<class 'list'>", [1]),
+        ],
+    )
+    def test_resolve_scalar_dtype(self, dtype, val):
+        """
+        Tests dtypes.resolve_scalar_dtype method
+
+        :return: None
+        :raise: AssertionError if 1.. test cases fail
+        """
+        assert dtype == dtypes.resolve_scalar_dtype(val)
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_pdarrays_datatypes(self, size):
+        assert dtypes.dtype("float64") == ak.ones(size).dtype
+        assert dtypes.dtype("str") == ak.array(["string {}".format(i) for i in range(0, size)]).dtype
```

Comment:
should we test more dtypes? like `uint`, `int`, and `bigint`? Also this is just be particular but using f strings is a cleaner way to declare the strings array
```python
ak.array([f"string {i}" for i in range(size)]).dtype
```

Reply:
Updated (with the exception of bigint)